test/suites/integration/https_gateway_listener_test.go

```diff
@@ -0,0 +1,81 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/aws/aws-application-networking-k8s/pkg/latticestore"
+	"github.com/aws/aws-application-networking-k8s/test/pkg/test"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+var _ = Describe("Test Https gateway listener", func() {
+	It("Create a Gateway with https listener, create a HttpRoute that reference it, the traffic should work", func() {
+		gateway := testFramework.NewHttpsGateway()
+
+		clientDeployment, clientService := testFramework.NewHttpApp(test.HTTPAppOptions{Name: "client"})
+		serverDeployment, serverService := testFramework.NewHttpApp(test.HTTPAppOptions{Name: "https-gw-server"})
+		httproute := testFramework.NewPathMatchHttpRoute(gateway, []client.Object{serverService}, "https")
+
```

Comment:
to pass "https" in the NewPathMatchHttpRoute(), it will refer the  "https" section in the `gateway ` listener